spec/lib/second_factor/actions/grant_admin_spec.rb

```diff
@@ -53,6 +53,25 @@ def create_instance(user, request = nil)
       )
     end
 
+    it "respects allowed_unicode_username_characters in username" do
+      begin
+        SiteSetting.unicode_usernames = true
+        SiteSetting.allowed_unicode_username_characters = "[äöüßÄÖÜẞ]"
+
+        Rails.application.reload_routes!
```

Comment:
It is necessary because otherwise RouteFormat class is cached and spec is flaky

Reply:
Ahh you are right! I wondered why `/admin/users/1191/Anführerin` is working. It is because RouteFormat.username allows '%' and browser is encoding that url to `/admin/users/1191/Anf%25C3%25BChrerin`. Therefore solution is much simpler! :)